source/credentials_provider_process.c

```diff
@@ -0,0 +1,374 @@
+};
+
+AWS_STATIC_STRING_FROM_LITERAL(s_access_key_id_name, "AccessKeyId");
+AWS_STATIC_STRING_FROM_LITERAL(s_secret_access_key_name, "SecretAccessKey");
+AWS_STATIC_STRING_FROM_LITERAL(s_session_token_name, "Token");
+AWS_STATIC_STRING_FROM_LITERAL(s_creds_expiration_name, "Expiration");
+
+static struct aws_credentials *s_parse_credentials_from_json_document(
+    struct aws_allocator *allocator,
+    struct aws_byte_buf *document) {
+
+    struct aws_credentials *credentials = NULL;
+    cJSON *document_root = NULL;
+    bool success = false;
+    bool parse_error = true;
+
+    document_root = cJSON_Parse((const char *)document->buffer);
+    if (document_root == NULL) {
+        AWS_LOGF_ERROR(AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse process output as Json document.");
+        goto done;
+    }
+
+    /*
+     * Pull out the three credentials components
+     */
+    cJSON *access_key_id = cJSON_GetObjectItemCaseSensitive(document_root, aws_string_c_str(s_access_key_id_name));
+    if (!cJSON_IsString(access_key_id) || (access_key_id->valuestring == NULL)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse AccessKeyId from process output Json document.");
+        goto done;
+    }
+
+    cJSON *secret_access_key =
+        cJSON_GetObjectItemCaseSensitive(document_root, aws_string_c_str(s_secret_access_key_name));
+    if (!cJSON_IsString(secret_access_key) || (secret_access_key->valuestring == NULL)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse SecretAccessKey from process output Json document.");
+        goto done;
+    }
+
+    cJSON *session_token = cJSON_GetObjectItemCaseSensitive(document_root, aws_string_c_str(s_session_token_name));
+    if (!cJSON_IsString(session_token) || (session_token->valuestring == NULL)) {
+        AWS_LOGF_ERROR(AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse Token from process output Json document.");
+        goto done;
+    }
+
+    cJSON *creds_expiration =
+        cJSON_GetObjectItemCaseSensitive(document_root, aws_string_c_str(s_creds_expiration_name));
+    if (!cJSON_IsString(creds_expiration) || (creds_expiration->valuestring == NULL)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER, "Failed to parse Expiration from process output Json document.");
+        goto done;
+    }
+
+    /*
+     * Build the credentials
+     */
+    struct aws_byte_cursor access_key_id_cursor = aws_byte_cursor_from_c_str(access_key_id->valuestring);
+    struct aws_byte_cursor secret_access_key_cursor = aws_byte_cursor_from_c_str(secret_access_key->valuestring);
+    struct aws_byte_cursor session_token_cursor = aws_byte_cursor_from_c_str(session_token->valuestring);
+    struct aws_byte_cursor creds_expiration_cursor = aws_byte_cursor_from_c_str(creds_expiration->valuestring);
+
+    if (access_key_id_cursor.len == 0 || secret_access_key_cursor.len == 0 || session_token_cursor.len == 0) {
```

Comment:
The SEP requires session token?  That's unexpected.

Reply:
you are right, it's not required.